src/org/omegat/languagetools/LanguageDataBroker.java

```diff
@@ -0,0 +1,215 @@
+
+    public LanguageDataBroker() {
+        this(ResourceDataBroker.RESOURCE_DIR, ResourceDataBroker.RULES_DIR);
+    }
+
+    @Override
+    public @NotNull URL getFromResourceDirAsUrl(String path) {
+        String completePath = getCompleteResourceUrl(path);
+        URL resource = getAsURL(completePath);
+        return Objects.requireNonNull(resource);
+    }
+
+    @Override
+    public @NotNull List<URL> getFromResourceDirAsUrls(String path) {
+        String completePath = getCompleteResourceUrl(path);
+        List<URL> resources = getAsURLs(completePath.substring(1));
+        return Objects.requireNonNull(resources);
+    }
+
+    private String getCompleteRulesUrl(String path) {
+        return this.appendPath(this.rulesDir, path);
+    }
+
+    private String getCompleteResourceUrl(String path) {
+        return this.appendPath(this.resourceDir, path);
+    }
+
+    @Override
+    public boolean resourceExists(String path) {
+        String completePath = this.getCompleteResourceUrl(path);
+        return this.getAsURL(completePath) != null;
+    }
+
+    private String appendPath(String baseDir, String path) {
+        StringBuilder completePath = new StringBuilder(baseDir);
+        if (!rulesDir.endsWith("/") && !path.startsWith("/")) {
+            completePath.append('/');
+        }
+
+        if (rulesDir.endsWith("/") && path.startsWith("/") && path.length() > 1) {
+            completePath.append(path.substring(1));
+        } else {
+            completePath.append(path);
+        }
+
+        return completePath.toString();
+    }
+
+    @Override
+    public boolean ruleFileExists(String path) {
+        String completePath = this.getCompleteRulesUrl(path);
+        return this.getAsURL(completePath) != null;
+    }
+
+    @Override
+    public @NotNull InputStream getFromResourceDirAsStream(String path) {
+        String completePath = getCompleteResourceUrl(path);
+        return Objects.requireNonNull(getAsStream(completePath));
+    }
+
+    @Override
+    public @NotNull List<String> getFromResourceDirAsLines(String path) {
+        try (InputStream inputStram = getFromResourceDirAsStream(path);
+                InputStreamReader inputStreamReader = new InputStreamReader(inputStram,
+                        StandardCharsets.UTF_8);
+                BufferedReader bufferedReader = new BufferedReader(inputStreamReader)) {
+            return bufferedReader.lines().collect(Collectors.toList());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Language getLanguage() {
+        org.omegat.util.Language omLang = LanguageManager.getProjectTargetLanguage();
+        if (omLang == null) {
+            return null;
+        }
+        Language lang = LanguageManager.getLTLanguage(omLang.getLanguageCode(), omLang.getCountryCode());
+        if (lang == null) {
+            lang = LanguageManager.getLTLanguage("en", "US");
+        }
+        return lang;
+    }
```

Comment:
This `LanguageDataBroker#getLanguage` can be move to `LanguageManager` class because all the operation depends on `LanguageManager`.